src/python/pants/option/arg_splitter.py

```diff
@@ -205,13 +205,17 @@ def assign_flag_to_scope(flg: str, default_scope: str) -> None:
     )
 
   @staticmethod
-  def spec(arg: str) -> bool:
-    """Does arg 'look like' a Spec (rather than a goal name).
+  def likely_a_spec(arg: str) -> bool:
+    """Return whether `arg` looks like a spec, rather than a goal name.
 
     An arg is a spec if it looks like an AddressSpec or a FilesystemSpec.
     In the future we can expand this heuristic to support other kinds of specs, such as URLs.
     """
-    return os.path.sep in arg or ':' in arg or '*' in arg or Path(arg).exists()
+    return (
+      any(symbol in arg for symbol in (os.path.sep, ':', '*'))
```

Comment:
The main reason I was thinking a regex might be useful is that it will be much faster than checking each symbol at once. If this ends up becoming hot on any profiles, we might want to consider converting it to a regex for performance. For now, I think this solution is the most clear, and that that's what we care about more here.

Reply:
I didn't profile here, but allegedly regex is seen as slower as `x in xs` in Python!  https://stackoverflow.com/questions/4901523/whats-a-faster-operation-re-match-search-or-str-find

Reply:
That's correct, but in doing three separate checks here, I strongly suspect (benchmarking always necessary) that you would be losing to a single regex match.